fuzz/src/onion_message.rs

```diff
@@ -209,6 +207,9 @@ mod tests {
 			let log_entries = logger.lines.lock().unwrap();
 			assert_eq!(log_entries.get(&("lightning::onion_message::messenger".to_string(),
 						"Received an onion message with path_id None and no reply_path".to_string())), Some(&1));
+			// TODO: Add reply path to one_hop_om instead?
```

Comment:
Sorry for the delay. I just wrote a test with hardcoded keys + printed out the encoded onion message, I think. Here's the branch I used if it helps: https://github.com/valentinewallace/rust-lightning/tree/2022-10-GEN-OM-FUZZ

Reply:
Seeing a test failure when cherrypicking this code. I pinged @valentinewallace offline but may need to wait for timezones to catch up.

Reply:
Looked into this. For the test where we're sending directly to a blinded path, we spuriously advance the blinded path on send because in fuzzing all the node pks are the same (so we think we're the intro node to the blinded path).

Reply:
Thanks for helping track this down! I've updated the fuzzer such that each node uses a different public key and was able to generate updated onion messages. Added a separate commit for this.

Also, I updated the fuzzer to test the code path for responding given a reply path. The least intrusive way of doing this was to have the unblinded one-hop case include a one-hop blinded path back. This required modifying `BlindedPath::new_for_message` to allow such paths. This branch contains the needed modifications for updating the generated onion messages: https://github.com/jkczyz/rust-lightning/tree/2023-02-offer-flow-fuzz